models/articles.models.js

```diff
@@ -10,14 +10,26 @@ exports.fetchArticleById = (article_id) => {
     })
 }
 
-exports.fetchAllArticles = () => {
-    return db.query(`SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count FROM articles
-    LEFT JOIN comments on comments.article_id = articles.article_id
-    GROUP BY articles.article_id
-    ORDER BY articles.created_at DESC;`)
-    .then(({ rows }) => {
-        return rows
-    })
+exports.fetchAllArticles = (topic) => {
+    if (!topic) {
+        return db.query(`SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count FROM articles
+        LEFT JOIN comments on comments.article_id = articles.article_id
+        GROUP BY articles.article_id
+        ORDER BY articles.created_at DESC;`)
+        .then(({ rows }) => {
+            return rows
+        })
+    } else {
+        return db.query(`SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count FROM articles
+        LEFT JOIN comments on comments.article_id = articles.article_id
+        WHERE articles.topic = $1
+        GROUP BY articles.article_id
+        ORDER BY articles.created_at DESC;`,
+        [topic])
+        .then(({ rows }) => {
+            return rows
```

Comment:
To make this code a bit more DRY, we should have one query string, which we conditionally add the query into if the query is defined.